File: snn/lif_neuron/lif_test.py
from lif_neuron import LeakyIntegrateFireNeuron

input_val = 0.3
spike_count = 0
neuron = LeakyIntegrateFireNeuron()
iter_count = 1

for i in range(30 * iter_count):
    if neuron.process(input_val) == 25:
        spike_count += 1

print("Input - ", input_val, " - Spike count - ", spike_count / iter_count)

File: snn/lif_neuron/lif_test.cpp
#include <iostream>
#include "lif_neuron.h"

using namespace std;

int main() {
	double input = 0.3;
	int spike_count = 0;
	LeakyIntegrateFireNeuron neuron;
	int iter = 1;
	for(int i = 0; i < 30 * iter; i++) {
		if(neuron.process(input) == 25) {
			spike_count++;
		}
	}
	cout << "Input - " << input << " - Spike count - " << double(spike_count / iter) << endl;
	return 0;
}